src/components/Sidebar/SidebarItem.tsx

```diff
@@ -35,12 +37,61 @@ const SidebarItemName = styled.h4`
   color: #1c1c1e;
 `;
 
-const SidebarItem: FC<MenuItemsType> = memo(({icon, title}) => {
+const ItemsWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const SidebarItemArrowIcon = styled.span`
+  margin-left: 51px;
+  width: 15px;
+  height: 15px;
+  & > svg {
+    height: 100%;
+    width: 100%;
+  }
+  & > strong {
+    & > svg {
+      transform: rotate(180deg);
+      height: 100%;
+      width: 100%;
+    }
+  }
+`;
+
+interface MenuItemPropTypes {
+  menuItem: MenuItemsType;
+  level?: number;
+}
+
+const SidebarItem: FC<MenuItemPropTypes> = memo(({menuItem, level = 0}) => {
+  const {icon, title, children} = menuItem;
+  const [isOpened, setIsOpened] = useState(false);
+  const settingsToogler = useCallback(() => {
+    setIsOpened(!isOpened);
+  }, [isOpened]);
+
   return (
-    <SidebarItemWrapper>
-      <SidebarItemIcon>{icon}</SidebarItemIcon>
-      <SidebarItemName>{title}</SidebarItemName>
-    </SidebarItemWrapper>
+    <ItemsWrapper style={{paddingLeft: `${48 * level}px`}}>
+      <SidebarItemWrapper onClick={children && settingsToogler}>
+        <SidebarItemIcon>{icon}</SidebarItemIcon>
+        <SidebarItemName>{title}</SidebarItemName>
+        {children && (
+          <SidebarItemArrowIcon>
+            {isOpened ? (
+              <strong>
+                <Arrow />
+              </strong>
+            ) : (
+              <Arrow />
+            )}
+          </SidebarItemArrowIcon>
+        )}
+      </SidebarItemWrapper>
+      {children &&
```

Comment:
isOpened && children?.map()

you check if children exists (not null or undefined) and if it does exist, map them